scripts/mk-initrd.sh

```diff
@@ -8,24 +19,36 @@ INITRD_DIR=${BUILD}/initrd
 
 rm -rf ${INITRD_DIR}/{usr,init}
 mkdir -p ${INITRD_DIR}/usr/{bin,share/ros}
+mkdir -p ${INITRD_DIR}/var/lib/system-docker
 
-cp -rf ${BUILD}/kernel/lib ${INITRD_DIR}/usr
-cp assets/docker           ${INITRD_DIR}/usr/bin/docker
-cp ${BUILD}/images.tar     ${INITRD_DIR}/usr/share/ros
-cp os-config.yml           ${INITRD_DIR}/usr/share/ros/
-cp bin/rancheros           ${INITRD_DIR}/usr/bin/ros
+cp -rf ${BUILD}/kernel/lib ${INITRD_DIR}/usr/
+if [ "$IS_ROOTFS" == "0" ]; then
+  cp ${BUILD}/images.tar     ${INITRD_DIR}/usr/share/ros/
+fi
+cp os-config${suffix}.yml  ${INITRD_DIR}/usr/share/ros/os-config.yml
+cp bin/ros                 ${INITRD_DIR}/usr/bin/
 ln -s usr/bin/ros          ${INITRD_DIR}/init
 ln -s bin                  ${INITRD_DIR}/usr/sbin
+ln -s usr/sbin             ${INITRD_DIR}/sbin
+
+DFS_ARCH=$(docker create ${DFS_ARCH_IMAGE})
+trap "docker rm -fv ${DFS_ARCH}" EXIT
 
-DFS=$(docker create ${DFS_IMAGE})
-trap "docker rm -fv ${DFS}" EXIT
-docker export ${DFS} | tar xvf - -C ${INITRD_DIR}  --exclude=usr/bin/dockerlaunch \
-                                                   --exclude=usr/bin/docker       \
-                                                   --exclude=usr/share/git-core   \
-                                                   --exclude=usr/bin/git          \
-                                                   --exclude=usr/bin/ssh          \
-                                                   --exclude=usr/libexec/git-core \
-                                                   usr
+docker export ${DFS_ARCH} | tar xvf - -C ${INITRD_DIR} --exclude=usr/bin/dockerlaunch \
+                                                       --exclude=usr/share/git-core   \
+                                                       --exclude=usr/bin/git          \
+                                                       --exclude=usr/bin/ssh          \
+                                                       --exclude=usr/libexec/git-core \
+                                                       usr
+
+if [ "$IS_ROOTFS" == "1" ]; then
+  DFS=$(docker run -d --privileged -v /lib/modules/$(uname -r):/lib/modules/$(uname -r) ${DFS_IMAGE})
```

Comment:
The build runs on a "normal" amd64 host :smile:  That's why we can create arm (and potentially other architectures) containers, but not run them during the build.